client/src/lib/solana/sol-data.ts

```diff
@@ -189,42 +248,52 @@ export class SolData extends Assignable {
     return VERSION;
   }
 
-  static parseDIDReferenceArray(
+  static async parseDIDReferenceArray(
     dids: (string | DIDVerificationMethod)[] | undefined
-  ): string[] {
-    return DecentralizedIdentifier.parseMaybeArray(dids).map(
+  ): Promise<(string | undefined)[]> {
```

Comment:
under what conditions is the result undefined? Shouldn't we filter them out then? An array of string | undefined is not a particularly nice value to have to deal with.

Reply:
That's true, I'll filter the undefineds here